include/swift/ClangImporter/ClangImporter.h

```diff
@@ -38,10 +38,11 @@ namespace clang {
   class Sema;
   class TargetInfo;
   class VisibleDeclConsumer;
+  class DeclarationName;
 }
 
 namespace swift {
-
+namespace version { class Version; }
```

Comment:
I don't think you can forward-declare these when you use them as values. We're just getting lucky with header import order. I suggest using a const reference for clang::DeclarationName and just including Version.h for swift::version::Version.